nifi-nar-bundles/nifi-pulsar-bundle/nifi-pulsar-processors/src/main/java/org/apache/nifi/processors/pulsar/pubsub/ConsumePulsar_1_X.java

```diff
@@ -0,0 +1,206 @@
+                consume(context, session);
+            }
+        } catch (PulsarClientException e) {
+            getLogger().error("Unable to consume from Pulsar Topic ", e);
+            context.yield();
+            throw new ProcessException(e);
+        }
+
+    }
+
+    private void handleAsync(ProcessContext context, ProcessSession session) {
+
+        try {
+            Future<Message> done = consumerService.take();
+            Message msg = done.get();
+
+            if (msg != null) {
+                FlowFile flowFile = null;
+                final byte[] value = msg.getData();
+                if (value != null && value.length > 0) {
+                    flowFile = session.create();
+                    flowFile = session.write(flowFile, out -> {
+                        out.write(value);
+                    });
+
+                   session.getProvenanceReporter().receive(flowFile, "From " + getWrappedConsumer(context).getTransitURL());
+                   session.transfer(flowFile, REL_SUCCESS);
+                   session.commit();
+                   getWrappedConsumer(context).getConsumer().acknowledgeAsync(msg);
+                }
+            }
+
+        } catch (InterruptedException | ExecutionException | PulsarClientException e) {
+            getLogger().error("Trouble consuming messages ", e);
+        }
+
+    }
+
+    @OnStopped
+    public void close(final ProcessContext context) {
+
+        getLogger().info("Disconnecting Pulsar Consumer");
+        if (consumer != null) {
+
+            context.getProperty(PULSAR_CLIENT_SERVICE)
+                .asControllerService(PulsarClientPool.class)
+                .getConsumerPool().evict(consumer);
+        }
+
+        consumer = null;
+    }
+
+    /*
+     * When this Processor expects to receive many small files, it may
+     * be advisable to create several FlowFiles from a single session
+     * before committing the session. Typically, this allows the Framework
+     * to treat the content of the newly created FlowFiles much more efficiently.
+     */
+    private void consume(ProcessContext context, ProcessSession session) throws PulsarClientException {
+
+        Consumer consumer = getWrappedConsumer(context).getConsumer();
+
+        final ComponentLog logger = getLogger();
+        final Message msg;
+        FlowFile flowFile = null;
+
+        try {
+
+            msg = consumer.receive();
+            final byte[] value = msg.getData();
+
+            if (value != null && value.length > 0) {
+                flowFile = session.create();
+                flowFile = session.write(flowFile, out -> {
+                    out.write(value);
+                });
+
+                session.getProvenanceReporter().receive(flowFile, "From " + context.getProperty(TOPIC).getValue());
+                session.transfer(flowFile, REL_SUCCESS);
+                logger.info("Created {} from {} messages received from Pulsar Server and transferred to 'success'",
```

Comment:
Let's do this as debug and wrap with an if statement because this will get called a lot.